binding/flutter/lib/porcupine.dart

```diff
@@ -92,121 +106,116 @@ class Porcupine {
-      _resourcesExtracted = true;
+    if (accessKey == null || accessKey == "") {
+      throw PorcupineInvalidArgumentException("'accessKey' must be set.");
     }
 
     if (modelPath == null || modelPath == "") {
       modelPath = _defaultModelPath;
     }
 
-    if (!await File(modelPath!).exists()) {
+    if (!(await File(modelPath!).exists())) {
       try {
         modelPath = await _extractResource(modelPath);
-      } catch (_) {
-        throw new PvArgumentError(
-            "Could not find model file at path '$modelPath'. If this is a packaged asset, ensure you have added it to your pubspec.yaml file.");
-      }
-    }
-
-    if (keywords != null && keywordPaths != null) {
-      throw new PvArgumentError(
-          "Both 'keywords' and 'keywordPaths' were set. Only one of the two arguments may be set for initialization.");
+      } catch (e) {/* don't do anything on fail */}
     }
 
-    if (keywordPaths == null || keywordPaths.length == 0) {
-      if (keywords == null || keywords.length == 0) {
-        throw new PvArgumentError(
-            "No built-in or custom keywords provided. Either 'keywords' or 'keywordPaths' must be set.");
-      }
-
-      keywordPaths = List.filled(keywords.length, "");
-      for (var i = 0; i < keywords.length; i++) {
-        if (BUILT_IN_KEYWORDS.contains(keywords[i])) {
-          keywordPaths[i] = _builtInKeywordPaths?[keywords[i]];
-        } else {
-          throw new PvArgumentError(
-              "The keyword `${keywords[i]}` is not a valid built-in keyword. Available built-in keywords are: $BUILT_IN_KEYWORDS");
-        }
-      }
+    if (keywordPaths == null || keywordPaths.isEmpty) {
+      throw PorcupineInvalidArgumentException("'keywordPaths' must be set.");
     }
 
     for (var i = 0; i < keywordPaths.length; i++) {
       if (keywordPaths[i] == null || keywordPaths[i] == "") {
-        throw new PvArgumentError(
+        throw PorcupineInvalidArgumentException(
             "One of the provided keyword paths was empty.");
       }
 
-      if (!await File(keywordPaths[i]!).exists()) {
+      if (!(await File(keywordPaths[i]!).exists())) {
         try {
           keywordPaths[i] = await _extractResource(keywordPaths[i]!);
-        } catch (_) {
-          throw new PvArgumentError(
-              "Could not find keyword file at path '${keywordPaths[i]}'. If this is a packaged asset, ensure you have added it to your pubspec.yaml file.");
-        }
+        } catch (e) {/* don't do anything on fail */}
       }
     }
 
-    if (sensitivities == null || sensitivities.length == 0) {
+    if (sensitivities == null || sensitivities.isEmpty) {
       sensitivities = List.filled(keywordPaths.length, 0.5);
     } else {
       for (double sensitivity in sensitivities) {
         if (sensitivity < 0 || sensitivity > 1 || sensitivity == double.nan) {
-          throw new PvArgumentError(
+          throw PorcupineInvalidArgumentException(
```

Comment:
I actually wonder if we should move all validation to native and just have the Flutter plugin be a super thin wrapper...